QGL/PulsePrimitives.py

```diff
@@ -74,38 +74,61 @@ def Id(channel, *args, **kwargs):
 
 # the most generic pulse is Utheta
 def Utheta(qubit,
-           amp=0,
+           angle=0,
            phase=0,
            label='Utheta',
            ignoredStrParams=[],
            **kwargs):
-    '''  A generic rotation with variable amplitude and phase. '''
+    '''
+    A generic rotation with variable angle and phase.
+    This primitive needs to convert the requested rotation angle into a Pulse
+    amplitude. If the user would like to specify a specific amplitude, he/she
+    can specify it directly via the 'amp' keyword argument.
+    '''
     params = overrideDefaults(qubit, kwargs)
-        #amp and phase are now pulse parameters rather than shape parameters
+    # amp and phase are now pulse parameters rather than shape parameters
     if "amp" in params:
-      del params["amp"]
+        del params["amp"]
     if "phase" in params:
-      del params["phase"]
+        del params["phase"]
+    # allow override of angle -> amplitude lookup if the user provides an "amp"
+    # keyword argument
+    if "amp" in kwargs:
+        amp = kwargs["amp"]
+    else:
+        # construct an angle -> amplitude lookup table
+        # TODO should this live in the Channel object instead?
+        angle2amp = {
+            pi    :  qubit.pulseParams['piAmp'],
+            -pi   : -qubit.pulseParams['piAmp'],
+            pi/2  :  qubit.pulseParams['pi2Amp'],
+            -pi/2 : -qubit.pulseParams['pi2Amp'],
+        }
+        if angle in angle2amp:
+            amp = angle2amp[angle]
+        else:
+            # linearly scale based upon the 'pi/2' amplitude
+            amp  = (angle / pi/2) * qubit.pulseParams['pi2Amp']
```

Comment:
I wonder if using some average between `qubit.pulseParams['pi2Amp']` and `qubit.pulseParams['piAmp']` may help the initial estimate for `pi/2 < amp < pi`.